src/user/dto/create-user.dto.ts

```diff
@@ -0,0 +1,12 @@
+import { IsString, IsInt, IsOptional } from 'class-validator';
```

Comment:
We should use command query responsibility pattern(https://docs.nestjs.com/recipes/cqrs).
This interface should be named CreateUserCommand.